File: lib/models/question.dart
import 'package:ecommerceapp/models/user.dart';

class Question {
  String id;
  String userId;
  String userName;
  String title;
  String body;
  String createdDate;
  int responseCount;
  int upvotes;

  Question(
      {required this.id,
      required this.userId,
      required this.userName,
      required this.title,
      required this.body,
      required this.createdDate,
      this.responseCount = 1,
      this.upvotes = 1});

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      id: json['id'] as String,
      userId: json['userId'] as String,
      userName: "Loading...",
      title: json['title'] as String,
      body: json['body'] as String,
      createdDate: json['date'] as String,
      upvotes: json['upvotes'] as int,
    );
  }
}

File: backend/Models/Question.cs
namespace ecommerceapp.models;
public class Question {
    public string Id { get; set; }
    public string User { get; set; }
    public string Title { get; set; }
    public string Body { get; set; }
    public string createdDate { get; set; }
    public List<Response> Responses { get; set; }
    public int Upvotes;

    public Question(string Id, string User, string Title, string Body, string createdDate, List<Response> Responses, int Upvotes) {
        this.Id = Id;
        this.User = User;
        this.Title = Title;
        this.Body = Body;
        this.createdDate = createdDate;
        this.Responses = Responses.ToList();
        this.Upvotes = Upvotes;
    }
}
